pycaching/geocaching.py

```diff
@@ -239,116 +239,49 @@ def get_logged_user(self, login_page=None):
-            rows = geocaches_table.find_all("tr")
-
-            # leave loop if there are no (more) results
-            if not rows:
-                return
-
-            # prepare language-dependent mappings
-            if start_index == 0:
-                cache_sizes_filter_wrapper = whole_page.find("div", class_="cache-sizes-wrapper")
-                localized_size_mapping = {
-                    # key = "Small" (localized), value = Size.small
-                    label.find("span").text.strip(): Size.from_number(label.find("input").get("value"))
-                    for label in cache_sizes_filter_wrapper.find_all("label")
-                }
-
-            # parse caches in result
-            for start_index, row in enumerate(rows, start_index):
-
-                limit -= 1  # handle limit
-                if limit < 0:
-                    return
-
-                # parse raw data
-                cache_details = row.find("span", "cache-details").text.split("|")
-                wp = cache_details[1].strip()
-
-                # create and fill cache object
-                # values are sanitized and converted in Cache setters
-                c = Cache(self, wp)
-                c.type = cache_details[0]
-                c.name = row.find("span", "cache-name").text
-                badge = row.find("svg", class_="badge")
-                c.found = "found" in str(badge) if badge is not None else False
-                c.favorites = row.find(attrs={"data-column": "FavoritePoint"}).text
-                if not (row.get("class") and "disabled" in row.get("class")):
-                    c.status = Status.enabled
-                c.pm_only = row.find("td", "pm-upsell") is not None
-
-                if c.pm_only:
-                    # PM only caches doesn't have other attributes filled in
-                    yield c
-                    continue
-
-                c.size = localized_size_mapping[row.find(attrs={"data-column": "ContainerSize"}).text.strip()]
-                c.difficulty = row.find(attrs={"data-column": "Difficulty"}).text
-                c.terrain = row.find(attrs={"data-column": "Terrain"}).text
-                c.hidden = row.find(attrs={"data-column": "PlaceDate"}).text
-                c.author = row.find("span", "owner").text[3:]  # delete "by "
-
-                logging.debug("Cache parsed: {}".format(c))
-                yield c
-
-            start_index += 1
-
-    def _search_get_page(self, point, start_index):
-        """Return one page for standard search as class:`bs4.BeautifulSoup` object.
-
-        :param .geo.Point point: Search center point.
-        :param int start_index: Determines the page. If start_index is greater than zero, this
-            method will use AJAX andpoint which is much faster.
+    def search(
+        self,
+        point: Point,
+        per_query: int = 200,
+        sort_by: Union[str, SortOrder] = SortOrder.date_last_visited,
```

Comment:
My preference is against adding new parameters to this method – to keep it dead simple for basic users and onboarding purposes.

On the other hand, I see some benefit of unifying the param list with `search_rect`. In that case, I'd strongly prefer having `*` here (see my other comment).

Given I am not daily user of this library anymore, so I will let others decide here.

Reply:
I still didn't quite get how the asterisk works there 😄

However, as an active library user, I would say I like having those arguments there since they make the work with caches a little bit easier. Those other arguments are optional, so users don't need to specify them.

The function needs to get only specified the `point` parameter. (`geocaching.search(Point(49.73, 13.38))` would work without any problems)

So, I hope adding the mentioned asterisk would be fine — https://github.com/tomasbedrich/pycaching/pull/206/commits/ea4817471fe5dc1b930be429f4aeeff8e3bf4560

Reply:
Oh, now I get how it works :D
So, yeah, it makes sense now :)

Fixed in: https://github.com/tomasbedrich/pycaching/pull/206/commits/8c839f455920536d376fc9e8665663a19f3aa6f2. I've also reverted my changes in the test back to test whether positional arguments work: https://github.com/tomasbedrich/pycaching/pull/206/commits/cc0c92f1ef9504f079ed9cfe988a7e16a85caac7.